lib/waterline/adapter/dql.js

```diff
@@ -132,7 +131,12 @@ module.exports = {
    * @return {[type]}            [description]
    */
   findOne: function(criteria, cb) {
-
+    
+    //make shallow copy of criteria so original does not get modified
+    if (Array.isArray(criteria) || _.isObject(criteria)) {
```

Comment:
I think we can simplify this to `if (_.isObject(criteria)) {` given that `_.isObject([]) === true`, actually I think there is no problem to remove the `if` and go with just `criteria = _.clone(criteria);`. I've seen that being done in other places.